src/NLog/Targets/Wrappers/AsyncTargetWrapper.cs

```diff
@@ -173,10 +178,25 @@ public int QueueLimit
         [DefaultValue(5)]
         public int FullBatchSizeWriteLimit { get; set; }
 
+#if NET4_5 || NET4_0
+        /// <summary>
+        /// Gets or sets whether to use the locking queue, instead of a lock-free concurrent queue
+        /// </summary>
+        public bool ForceLockingQueue
+        {
+            get { return _requestQueue is AsyncRequestQueue; }
+            set
+            {
+                if (value != _requestQueue is AsyncRequestQueue)
```

Comment:
I prefer the pattern matching syntax her, eg `is AsyncRequestQueue queue`  (c# 7)

Reply:
Not sure I understand, but will add some parentheses.

Reply:
When pattern matching then the cast (line below) is not needed.

Reply:
Still don't understand